src/sqlancer/tidb/oracle/TiDBCERTOracle.java

```diff
@@ -0,0 +1,285 @@
+    }
+
+    @Override
+    public void check() throws SQLException {
+        queryPlan1Sequences = new ArrayList<>();
+        queryPlan2Sequences = new ArrayList<>();
+
+        // Randomly generate a query
+        TiDBTables tables = state.getSchema().getRandomTableNonEmptyTables();
+        gen = new TiDBExpressionGenerator(state).setColumns(tables.getColumns());
+        select = new TiDBSelect();
+
+        List<TiDBExpression> fetchColumns = new ArrayList<>();
+        fetchColumns.addAll(Randomly.nonEmptySubset(tables.getColumns()).stream().map(c -> new TiDBColumnReference(c))
+                .collect(Collectors.toList()));
+        select.setFetchColumns(fetchColumns);
+
+        List<TiDBExpression> tableList = tables.getTables().stream().map(t -> new TiDBTableReference(t))
+                .collect(Collectors.toList());
+        List<TiDBExpression> joins = TiDBJoin.getJoinsWithoutNature(tableList, state);
+        select.setJoinList(joins);
+        select.setFromList(tableList);
+        if (Randomly.getBoolean()) {
+            select.setWhereClause(gen.generateExpression());
+        }
+        if (Randomly.getBooleanWithRatherLowProbability()) {
+            select.setOrderByExpressions(gen.generateOrderBys());
+        }
+        if (Randomly.getBoolean()) {
+            select.setGroupByExpressions(select.getFetchColumns());
+            if (Randomly.getBoolean()) {
+                select.setHavingClause(gen.generateHavingClause());
+            }
+        }
+        if (Randomly.getBoolean()) {
+            select.setLimitClause(gen.generateExpression());
+        }
+        if (Randomly.getBoolean()) {
+            select.setOffsetClause(gen.generateExpression());
+        }
+
+        // Get the result of the first query
+        String queryString1 = TiDBVisitor.asString(select);
+        double rowCount1 = getRow(state, queryString1, queryPlan1Sequences);
+
+        // Mutate the query
+        boolean increase = false;
+        int mutator = (int) Randomly.getNotCachedInteger(0, 7);
+
+        switch (mutator) {
+        case 0:
+            increase = mutateJoin();
+            break;
+        case 1:
+            increase = mutateWhere();
+            break;
+        case 2:
+            increase = mutateGroupBy();
+            break;
+        case 3:
+            increase = mutateHaving();
+            break;
+        case 4:
+            increase = mutateAnd();
+            break;
+        case 5:
+            increase = mutateOr();
+            break;
+        case 6:
+            increase = mutateLimit();
+            break;
+        default:
+            throw new AssertionError();
+        }
+
+        // Get the result of the second query
+        String queryString2 = TiDBVisitor.asString(select);
+        double rowCount2 = getRow(state, queryString2, queryPlan2Sequences);
+
+        // Check structural equivalence
+        if (DBMSCommon.editDistance(queryPlan1Sequences, queryPlan2Sequences) > 1) {
```

Comment:
I guess we could just have a method `DBMSCommon.areQueryPlanSequencesSimilar` or so, which might be less error-prone to use? In that case, we wouldn't need to expose that we are using the edit distance here.

Reply:
I think edit distance is more concrete and can be used in other code?

Reply:
Given that you include a comment to explain what this does call does, I think it's clearer to just have a method with a similar name.

Reply:
We can remove the comment.

Reply:
We can remove the comment.
However, I think changing the function name makes it less general and clear.
Anyway, updated.